### ISSUE TITLE
HttpResponseException thrown from view results in ErrorException

### ISSUE THREAD

**Issue by acurrieclark** [livewire/livewire#1442]
Statamic 3 uses template tags to allow you to redirect the user. It does this by calling `abort(redirect($location, 302))` which produces an HttpResponseException.

As this hits the Livewire compiler engine, it is passed on to the the `PhpEngine` class which then wraps it in an `ErrorException` rather than producing a redirect.

Might it be possible to add `HttpResponseException` to the allow list of permitted exceptions?

Happy to put together a pull request if this makes sense.

**Comment by austenc** [comment]
@acurrieclark any chance you could provide a reproduction repo for this? Are you using the [Livewire Addon](https://statamic.com/addons/jonassiewertsen/livewire)?

If you can provide an easy way to reproduce, I'd be happy to take a look.

**Comment by acurrieclark** [comment]
@austenc no problem. Will put together a repo for you. I am using the addon, but at the time of creating this issue it didn't seem to help with this.

**Comment by calebporzio** [comment]
Hey @acurrieclark  - Ah that's interesting and makes sense why livewire would eat the redirect.

I'm open to a PR to add exceptions (I think there are internal exemptions, but we can add a public API for this) - Thanks!

Going to close this issue.

**Comment by acurrieclark** [comment]
Just to loop back round on this, in putting together a demo repo for @austenc I realised that if I altered the method of including my statamic templates, I could avoid this issue entirely.

I was including an `antlers` template using the blade `@include()` syntax, which meant that it was loaded through the blade compiler engine. By loading it properly with the Statamic `{!! $template_content !!}} tag, the issue disappears.

I have put together a pull request, which I will keep in the wings, just in case there is any further interest. For now it seems that a small change of tack has created a simpler solution, and explains why I can't find any further information on this issue elsewhere.

Thanks for the nudge in putting together a reproduction repo; i wouldn't have solved it without it!
